src/thor/costmatrix.cc

```diff
@@ -252,9 +253,10 @@ bool CostMatrix::SourceToTarget(Api& request,
       throw valhalla_exception_t{430};
     }
     // Allow this process to be aborted
-    if (interrupt_ && (n++ % kInterruptIterationsInterval) == 0) {
+    if (interrupt_ && (interrupt_n++ % kInterruptIterationsInterval) == 0) {
```

Comment:
needed to come up with `interrupt_n`: `n` is used to abort the expansion (among other things), but it was never incremented unless there was an interrupt. i.e. matrix is vulnerable to home-cooked DDOS right now.. this should fix it now.

Reply:
good catch! just a note: its still vulnerable (and all other endpoints) in the default configuration as well since we dont set a default timeout for the requests in the default config (i still think this is fine though and that we should not prescribe a timeout value)